File: algospot/Divide conquer/QUADTREE.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int start = 0;
char * makeReverse(char * zipFile){
	if(zipFile[start] == 'x'){
		start++;
		char * first = (char *)malloc(sizeof(char)* 1000);
		char * second = (char *)malloc(sizeof(char)* 1000);
		char * third = (char *)malloc(sizeof(char)* 1000);
		char * fourth = (char *)malloc(sizeof(char)* 1000);
		
		strcpy(first, makeReverse(zipFile));
		strcpy(second, makeReverse(zipFile));
	    strcpy(third, makeReverse(zipFile));
		strcpy(fourth, makeReverse(zipFile));

		static char reverseFile[1000];
		strcpy(reverseFile, "x");
		strcat(reverseFile, third);
		strcat(reverseFile, fourth);
		strcat(reverseFile, first);
		strcat(reverseFile, second);
		return reverseFile;
	}
	else{
		static char file[1]; 
		file[0] = zipFile[start];
		start++;
		return file;
	}
}

int main(){
	int caseNum;
	scanf("%d", &caseNum);
	char answer[caseNum][1000];
	
	for(int i = 0; i<caseNum; i++){
		char * zipFile;
		zipFile = (char*)malloc(sizeof(char)*1000);
		scanf("%s", zipFile);
        zipFile = makeReverse(zipFile);
		strcpy(answer[i], zipFile);
		start = 0;
	}
    
	for(int i = 0; i<caseNum; i++) printf("%s\n", answer[i]);
	return 0;
}

File: Divide conquer/QUADTREE.py
answer = []

check = 0

def reverseQuad(quad):
    global check

    if quad[check] == 'w':
        check += 1
        return "w"
    elif quad[check] == 'b':
        check += 1
        return "b"
    else:
        check += 1
        first = reverseQuad(quad)
        second = reverseQuad(quad)
        third = reverseQuad(quad)
        fourth = reverseQuad(quad)
        return "x" + third+fourth+first+second


case = int(input())
for i in range(case):
    quadTree = str(input())
    answer.append(reverseQuad(quadTree))
    check = 0

for i in range(case):
    print(answer[i])

